govulncheck CI action reports issues with paths on Windows

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The CI action [govulncheck](https://github.com/rook/rook/actions/workflows/golangci-lint.yaml) is failing across all branches starting today as seen in [this run](https://github.com/rook/rook/actions/runs/6807863557/job/18515045852?pr=13179):

```
Vulnerability #1: GO-2023-2186
    Incorrect detection of reserved device names on Windows in path/filepath
  More info: https://pkg.go.dev/vuln/GO-2023-2186
  Standard library
    Found in: path/filepath@go1.20.10
    Fixed in: path/filepath@go1.21.4
    Example traces found:
Error:       #1: pkg/operator/k8sutil/deployment.go:77:55: k8sutil.UpdateDeploymentAndWait calls patch.PatchMaker.Calculate, which eventually calls filepath.IsLocal

Vulnerability #2: GO-2023-2185
    Insecure parsing of Windows paths with a \??\ prefix in path/filepath
  More info: https://pkg.go.dev/vuln/GO-2023-2185
  Standard library
    Found in: path/filepath@go1.20.10
    Fixed in: path/filepath@go1.21.4
    Platforms: windows
    Example traces found:
Error:       #1: tests/framework/utils/exec_utils.go:101:17: utils.ExecuteCommand calls exec.Cmd.Start, which eventually calls filepath.Abs
Error:       #2: pkg/util/exec/exec.go:167:21: exec.CommandExecutor.ExecuteCommandWithCombinedOutput calls exec.Command, which calls filepath.Base
Error:       #3: pkg/daemon/ceph/client/config.go:289:40: client.WriteCephConfig calls filepath.Clean
Error:       #4: tests/framework/utils/helm_helper.go:180:29: utils.FindRookRoot calls filepath.Dir
Error:       #5: pkg/daemon/ceph/cleanup/hostpath.go:41:31: cleanup.cleanMonDirs calls filepath.Glob
Error:       #6: pkg/operator/k8sutil/deployment.go:77:55: k8sutil.UpdateDeploymentAndWait calls patch.PatchMaker.Calculate, which eventually calls filepath.IsLocal
Error:       #7: pkg/daemon/ceph/osd/daemon.go:498:72: osd.getAvailableDevices calls filepath.Join
Error:       #8: tests/framework/utils/exec_utils.go:101:17: utils.ExecuteCommand calls exec.Cmd.Start, which eventually calls filepath.VolumeName
Error:       #9: pkg/daemon/ceph/osd/kms/vault.go:100:21: kms.InitVault calls vault.New, which eventually calls filepath.Walk
```

**Expected behavior:**
The CI checks should pass. Either we need to suppress this issue since the operator does not run on windows, or we need a fix.

--- a/.github/workflows/golangci-lint.yaml
+++ b/.github/workflows/golangci-lint.yaml
@@ -49,5 +49,8 @@ jobs:
     name: govulncheck
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/setup-go@v4
+        with:
+         go-version: "1.21.4"
       - name: govulncheck
         uses: golang/govulncheck-action@v1

